File: soln.rb
def convert_to_24_hour_format(time)
    time_part, indicator = time.split(' ')
    hours, minutes = time_part.split(':').map(&:to_i)
  
    converted_hours = hours
    if indicator == 'PM' && hours != 12
      converted_hours += 12
    elsif indicator == 'AM' && hours == 12
      converted_hours = 0
    end
  
    formatted_time = "#{converted_hours.to_s.rjust(2, '0')}:#{minutes.to_s.rjust(2, '0')}"
    formatted_time
  end
  
  # Example usage
  input_time = '02:30 PM'
  converted_time = convert_to_24_hour_format(input_time)
  puts converted_time  # Output: "14:30"
  

File: soln.js
function convertTo24HourFormat(time) {
  // Parse the time string
  const [timePart, indicator] = time.split(" ");
  const [hours, minutes] = timePart.split(":").map(Number);

  // Convert hours to 24-hour format
  let convertedHours = hours;
  if (indicator === "PM" && hours !== 12) {
    convertedHours += 12;
  } else if (indicator === "AM" && hours === 12) {
    convertedHours = 0;
  }

  // Format the time in 24-hour format
  const formattedTime = `${String(convertedHours).padStart(2, "0")}:${String(
    minutes
  ).padStart(2, "0")}`;

  return `${time} => ${formattedTime} hrs`;
}

// Example usage
const inputTime = "12:30 AM";
const convertedTime = convertTo24HourFormat(inputTime);
console.log(convertedTime); // Output: "14:30"
